app/forms/instance-create.tsx

```diff
@@ -398,11 +429,15 @@ export function CreateInstanceForm() {
         defaultValue={defaultSource}
         onValueChange={(val) => {
           setValue('bootDiskSourceType', val as BootDiskSourceType)
-          // if the user switches to the disk tab, clear the selected image;
-          // if they switch to the same tab that has the defaultImage, select it
-          setValue('bootDiskSource', val === defaultSource ? defaultImage?.id : '')
-          // clear any form errors
-          form.clearErrors('bootDiskSource')
+          if (val === 'siloImage') {
+            setValue('bootDiskSource', siloImageSource)
+          }
+          if (val === 'projectImage') {
+            setValue('bootDiskSource', projectImageSource)
+          }
+          if (val === 'disk') {
+            setValue('bootDiskSource', bootDiskSource)
+          }
```

Comment:
Why do we need to keep the common `bootDiskSource` at all? Couldn't both the onSubmit boot disk logic and the size updater look at `siloImageSource`, etc. dynamically based on `bootDiskSourceType`?

Reply:
There was an issue I was seeing where some cached state wasn't clearing when changing tabs and so the intended source wasn't getting updated, but I'll look into this a bit more.